packages/backend/src/services/UserService.ts

```diff
@@ -488,6 +488,35 @@ export class UserService extends BaseService {
         }
     }
 
+    /**
+     * In this method we check if the user is allowed to register in this instance before registerUser happens.
+     * If the instance is a single org, and there are already users and orgs,
+     * and the user's email domain is allowed in the org whitelisted,
+     * we throw a ForbiddenError.
+     * @param inviteCode invite code
+     * @param email email for the user registering
+     */
+    async checkNewUserRegistrationAllowed(
+        inviteCode: string | undefined,
+        email: string,
+    ) {
+        if (
+            inviteCode === undefined &&
+            !this.lightdashConfig.allowMultiOrgs &&
+            (await this.userModel.hasUsers()) &&
+            (await this.organizationModel.hasOrgs()) &&
+            (
+                await this.organizationModel.getAllowedOrgsForDomain(
+                    getEmailDomain(email),
+                )
+            ).length !== 1
+        ) {
+            throw new ForbiddenError(
+                `You can't register a new user on this instance. Please contact your organization administrator and ask for an invite`,
```

Comment:
We need to include the email in here too. Some users have been confused why they don't have access - it could be because the email we got from their SSO provider doesn't match what they have in the database.

Reply:
![image](https://github.com/user-attachments/assets/4cfc69a2-3809-4cf0-b04d-939da6ba4291)
Done